blockchain/blockdao.go

```diff
@@ -123,13 +138,50 @@ func (dao *blockDAO) Start(ctx context.Context) error {
 			return errors.Wrap(err, "failed to write initial value for total actions")
 		}
 	}
+
+	err = dao.initStores()
+	if err != nil {
+		return err
+	}
+
 	value, _ := dao.kvstore.Get(blockNS, totalActionsKey)
 	totalActions := enc.MachineEndian.Uint64(value)
 	if totalActions != 0 {
 		return nil
 	}
+
 	return dao.countActions()
 }
+func (dao *blockDAO) initStores() error {
+	cfg := dao.cfg
+	model, dir := getFileNameAndDir(cfg.DbPath)
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return err
+	}
+	var maxN uint64
+	for _, file := range files {
```

Comment:
what if the directory mix some other files in? for example, we currently put chain.db and trie.db in the same dir

Reply:
only find files like this:chain-xxxxxxxx.db